File: src/main/java/com/enablehr/challenge/java/entity/Todo.java
package com.enablehr.challenge.java.entity;

import org.springframework.data.jpa.domain.AbstractPersistable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "todos")
public class Todo extends AbstractPersistable<Integer> {

  @Column(nullable = false)
  @Id
  private Integer id;

  @Column(nullable = false)
  private Boolean completed;

  @Column(nullable = false)
  private String text;

  @Column(nullable = false)
  private Integer parentId;

  @Override
  public Integer getId() {
    return id;
  }

  @Override
  public void setId(Integer id) {
    this.id = id;
  }

  public Boolean getCompleted() {
    return completed;
  }

  public void setCompleted(Boolean completed) {
    this.completed = completed;
  }

  public String getText() {
    return text;
  }

  public void setText(String text) {
    this.text = text;
  }

  public Integer getParentId() {
    return parentId;
  }

  public void setParentId(Integer parentId) {
    this.parentId = parentId;
  }
}

File: src/main/kotlin/com/enablehr/challenge/kotlin/entity/Todo.kt
package com.enablehr.challenge.kotlin.entity

import org.springframework.data.jpa.domain.AbstractPersistable

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "todos")
data class Todo (

    @field:Column(nullable = false)
    var completed: Boolean = false,

    @field:Column(nullable = false)
    var text: String = ""

) : AbstractPersistable<Int>()
